creator-node/src/routes/tracks.js

```diff
@@ -4,58 +4,73 @@ const { Buffer } = require('ipfs-http-client')
 
 const ffmpeg = require('../ffmpeg')
 const ffprobe = require('../ffprobe')
-
 const models = require('../models')
 const { saveFileFromBuffer, saveFileToIPFSFromFS, removeTrackFolder, trackFileUpload } = require('../fileManager')
-const { handleResponse, successResponse, errorResponseBadRequest, errorResponseServerError } = require('../apiHelpers')
+const { handleResponse, successResponse, errorResponseBadRequest, errorResponseServerError, errorResponseSocketTimeout } = require('../apiHelpers')
 const { getFileUUIDForImageCID } = require('../utils')
 const { authMiddleware, syncLockMiddleware, ensurePrimaryMiddleware, triggerSecondarySyncs } = require('../middlewares')
 
-module.exports = function (app) {
+const TRACK_CONTENT_SOCKET_TIMEOUT = 1000*60*20 // Default = 1,200,000ms = 20min
+// const TRACK_CONTENT_SOCKET_TIMEOUT = 1000 // Default = 1,200,000ms = 20min
+
+module.exports = function (app) {  
   /**
    * upload track segment files and make avail - will later be associated with Audius track
    * @dev - currently stores each segment twice, once under random file UUID & once under IPFS multihash
    *      - this should be addressed eventually
    */
-  app.post('/track_content', authMiddleware, ensurePrimaryMiddleware, syncLockMiddleware, trackFileUpload.single('file'), handleResponse(async (req, res) => {
-    if (req.fileFilterError) return errorResponseBadRequest(req.fileFilterError)
-
-    // create and save track file segments to disk
-    let segmentFilePaths
+  app.post('/track_content', authMiddleware, trackFileUpload.single('file'), handleResponse(async (req, res) => {
+    // Increase serverTimeout from default 2min to ensure large files can be uploaded.
+    req.setTimeout(TRACK_CONTENT_SOCKET_TIMEOUT)
     try {
-      req.logger.info(`Segmenting file ${req.fileName}...`)
-      const segmentTimeStart = Date.now()
-      segmentFilePaths = await ffmpeg.segmentFile(req, req.fileDir, req.fileName)
-      req.logger.info(`Segment file time: ${Date.now() - segmentTimeStart}ms`)
-    } catch (err) {
-      removeTrackFolder(req, req.fileDir)
-      return errorResponseServerError(err)
-    }
+      // Short-circuit processing after req has timed-out.
+      await new Promise((resolve, reject) => {
```

Comment:
can this be achieved without a promise? or is this to make sure it executes after timeout?

Reply:
like just calling setTimeout would do the same thing

Reply:
will discuss afk